### ISSUE TITLE
Support of gitmodules

### ISSUE THREAD

**Issue by SHoen** [opendevstack/ods-jenkins-shared-library#533]
**Is your feature request related to a problem? Please describe.**
I would like to use gitmodules to resolve and split dependencies which are used in multiple components.

**Describe the solution you'd like**
If .gitmodules are configured the jenkins should init and update them together with the sourcecode of the repo

**Describe alternatives you've considered**
No alternatives are considered 

**Additional context**
https://git-scm.com/docs/gitmodules#_examples

**Comment by serverhorror** [comment]
Strongly opposed to that. `subtree` is a superior alternative to submodules and will always work with a "simple" git clone

**Comment by SHoen** [comment]
I agree that if you just include some dependency subtree is the way to go. But if you also own the component and want also to push code back gitmodules are the better choice. 
I would say, depends on your usecase.
